beacon-chain/utils/shuffle.go

```diff
@@ -70,3 +79,80 @@ func SplitIndices(l []uint64, n uint64) [][]uint64 {
 	}
 	return divided
 }
+
+// PermutedIndex Return `p(index)` in a pseudorandom permutation `p` of `0...list_size - 1` with ``seed`` as entropy.
+//     Utilizes 'swap or not' shuffling
+//
```

Comment:
Please also add somewhere in this godoc that this implementation is based on the original implementation from protolambda, https://github.com/protolambda/eth2-shuffle

Reply:
important. done